java/code/src/com/redhat/rhn/frontend/action/systems/sdc/SystemOverviewAction.java

```diff
@@ -129,6 +129,10 @@ public ActionForward execute(ActionMapping mapping, ActionForm form,
             createErrorMessage(request, "packagelist.jsp.modulespresent", null);
         }
 
+        if (s.asMinionServer().isPresent() && s.asMinionServer().get().doesOsSupportsTransactionalUpdate()) {
+            createErrorMessage(request, "overview.jsp.transactionalupdate.reboot", null);
+        }
```

Comment:
Small suggestion if you want to be even _more functional_ and not use the `isPresent()` and the `get()` at the same time, but then I leave it up to you to take it or discard it:

```java
s.asMinionServer().ifPresent(minion -> {
    if (minion.doesOsSupportsTransactionalUpdate()) {
        createErrorMessage(request, "overview.jsp.transactionalupdate.reboot", null);
    }
});
```